webrtc/videoroom/lib/videoroom/participant_mapper.ex

```diff
@@ -0,0 +1,58 @@
+defmodule VideoRoom.ParticipantMapper do
```

Comment:
I am wondering about creating GenServer for storing association between `display_name` and `mids`. In fact there is the module [`Membrane.WebRTC.Endpoint`](https://github.com/membraneframework/membrane_webrtc_plugin/blob/master/lib/membrane_webrtc_plugin/endpoint.ex) which aggregates Endpoint's inbound tracks. It could be better to move `display_name` there. We could even create there something like `user_data` to make it more generic.

`Membrane.WebRTC.Endpoint` is integrated with videoroom on branch [`track-filter`](https://github.com/membraneframework/membrane_demo/blob/track-filter/webrtc/videoroom/lib/videoroom/pipeline.ex#L57)

Reply:
The reason behind putting that mapper in other genserver was to reduce pipeline's state manipulation complexity as adding another function modifying the state would just make everything worse. 

I wouldn't go for putting things such `display_name` in `Membrane.WebRTC.Endpoint` as it is low-level abstraction and it probably should not be aware of participant specific information. It doesn't event know about `peer_pid` which is only tracked in pipeline by monitoring the pid.

Reply:
I don't still think we need a separate proccess here. It looks more like we use GenServer for code organization. We will also distribute state among different repositories. The idea is not to store display_name directly in an Endpoint but something like `user_data` of type `any` so the API would look like this

```elixir
Endpoint.get_user_data/1
Endpoint.update_user_data/1
```

In such scenario we can associate any arbitrary data with the Endpoint and don't need to care about cleaning it when the Endpoint disconnects from server.

Reply:
Ok, seems reasonable but I would opt for different name as the name `user` does not fit this level of abstraction.

Reply:
And yet I find another problem with this approach. When we decide to keep such data in `Endpoint` we will end up with having to query each endpoint for its metadata. With current approach we keep everything in a single place and we are done with a single request to the gen server.

Reply:
> Ok, seems reasonable but I would opt for different name as the name user does not fit this level of abstraction.

`user_data` was an example we can name it `data` `ctx` `state` etc.

> And yet I find another problem with this approach. When we decide to keep such data in Endpoint we will end up with having to query each endpoint for its metadata. With current approach we keep everything in a single place and we are done within a single request to the gen server.

But it's a simple Enum.map operation, isn't it? We don't need to think about optimizing it at this stage

Reply:
> But it's a simple Enum.map operation, isn't it? We don't need to think about optimizing it at this stage
How is that a simple enum? Don't we have to query each endpiont which will end up in sending and receiving messages?

Reply:
I mean `Endpoin.t()` not `EndpointBin`. If we had `data` in [`Endpoint.t()`](https://github.com/membraneframework/membrane_webrtc_plugin/blob/d1286e593e00f78174ed056d73fda0df33afcb9d/lib/membrane_webrtc_plugin/endpoint.ex#L10-L14) we would do 
```elixir
Enum.map(endpoints, fn endpoint -> Endpoint.get_data(endpoint) end)
```